permuta/patterns/perm.py

```diff
@@ -1174,9 +1175,44 @@ def max_drop_size(self):
             >>> Perm((1, 0)).max_drop_size()
             1
             >>> Perm((2, 0, 1)).max_drop_size()
-            2"""
+            2
+        """
         return max((val - idx for idx, val in enumerate(self)), default=0)
 
+    def holeyness(self) -> int:
+        """The holeyness of a permutation.
+        See: https://www.findstat.org/StatisticsDatabase/St001469/
+        https://mathoverflow.net/questions/340179/how-rare-are-unholey-permutations
+
+        Examples:
+            >>> Perm((1, 0)).holeyness()
+            0
+            >>> Perm((0, 1, 2)).holeyness()
+            0
+            >>> Perm((0, 2, 1)).holeyness()
+            1
+            >>> Perm((1, 0, 2)).holeyness()
+            1
+        """
+
+        def delta(d_set):
+            return sum(1 for x in d_set if x + 1 not in d_set)
+
+        def set_generator(num):
+            yield ()
+            for y in range(1, num + 1):
+                for x in itertools.combinations(range(num), y):
+                    yield x
+
+        n = len(self)
+        holyness = 0
+        for tmp_set in set_generator(n):
+            h_pi = delta([self[s] for s in tmp_set])
+            h_s = delta(tmp_set)
+            holyness = max(holyness, h_pi - h_s)
```

Comment:
I think this would be more pythonic - also use sets since you are just iterating and checking if `not in` which is linear for list/tuples.

``` python
return max(
        _delta({perm[s] for s in tmp_set}) - _delta(set(tmp_set))
        for tmp_set in _set_generator(len(perm))
    )
 ```